src/core/lib/channel/message_size_filter.c

```diff
@@ -38,13 +38,16 @@
 #include <grpc/support/log.h>
 #include <grpc/support/string_util.h>
 
+#include "src/core/ext/client_config/method_config.h"
 #include "src/core/lib/channel/channel_args.h"
 
 #define DEFAULT_MAX_SEND_MESSAGE_LENGTH -1  // Unlimited.
 // The protobuf library will (by default) start warning at 100 megs.
 #define DEFAULT_MAX_RECV_MESSAGE_LENGTH (4 * 1024 * 1024)
 
 typedef struct call_data {
+  int max_send_size;
```

Comment:
These fields can actually be set from either channel args or from the service config, depending on which value is lower.  The channel arg values are of type `int`, so using `int32_t` would be just as "wrong" for channel args as `int` is for the service config values.

In practice, I think this is an academic issue, because `int32_t` and `int` should be the same type.